clang-tools-extra/clang-tidy/readability/UseSpanFirstLastCheck.cpp

```diff
@@ -18,108 +18,77 @@ using namespace clang::ast_matchers;
-  unsigned NumArgs = Call->getNumArgs();
-  if (NumArgs == 0 || NumArgs > 2)
-    return;
+  if (const auto *FirstCall =
+          Result.Nodes.getNodeAs<CXXMemberCallExpr>("first_subspan")) {
+    const auto *Count = Result.Nodes.getNodeAs<Expr>("count");
+    if (!Count)
+      return;
 
-  const Expr *Offset = Call->getArg(0);
-  const Expr *Count = NumArgs > 1 ? Call->getArg(1) : nullptr;
-  auto &Context = *Result.Context;
-
-  class SubspanVisitor : public RecursiveASTVisitor<SubspanVisitor> {
-  public:
-    SubspanVisitor(const ASTContext &Context) : Context(Context) {}
-
-    TraversalKind getTraversalKind() const {
-      return TK_IgnoreUnlessSpelledInSource;
-    }
-
-    bool VisitIntegerLiteral(IntegerLiteral *IL) {
-      if (IL->getValue() == 0)
-        IsZeroOffset = true;
-      return true;
-    }
-
-    bool VisitBinaryOperator(BinaryOperator *BO) {
-      if (BO->getOpcode() == BO_Sub) {
-        if (const auto *SizeCall = dyn_cast<CXXMemberCallExpr>(BO->getLHS())) {
-          if (SizeCall->getMethodDecl()->getName() == "size") {
-            IsSizeMinusN = true;
-            SizeMinusArg = BO->getRHS();
-          }
-        }
-      }
-      return true;
-    }
-
-    bool IsZeroOffset = false;
-    bool IsSizeMinusN = false;
-    const Expr *SizeMinusArg = nullptr;
-
-  private:
-    const ASTContext &Context;
-  };
-
-  SubspanVisitor Visitor(Context);
-  Visitor.TraverseStmt(const_cast<Expr *>(Offset->IgnoreImpCasts()));
-
-  // Build replacement text
-  std::string Replacement;
-  if (Visitor.IsZeroOffset && Count) {
-    // subspan(0, count) -> first(count)
-    const StringRef CountStr = Lexer::getSourceText(
+    StringRef CountText = Lexer::getSourceText(
         CharSourceRange::getTokenRange(Count->getSourceRange()),
-        Context.getSourceManager(), Context.getLangOpts());
-    const auto *Base =
-        cast<CXXMemberCallExpr>(Call)->getImplicitObjectArgument();
-    const StringRef BaseStr = Lexer::getSourceText(
-        CharSourceRange::getTokenRange(Base->getSourceRange()),
-        Context.getSourceManager(), Context.getLangOpts());
-    Replacement = BaseStr.str() + ".first(" + CountStr.str() + ")";
-  } else if (Visitor.IsSizeMinusN && Visitor.SizeMinusArg) {
-    // subspan(size() - n) -> last(n)
-    const StringRef ArgStr = Lexer::getSourceText(
-        CharSourceRange::getTokenRange(Visitor.SizeMinusArg->getSourceRange()),
-        Context.getSourceManager(), Context.getLangOpts());
-    const auto *Base =
-        cast<CXXMemberCallExpr>(Call)->getImplicitObjectArgument();
-    const StringRef BaseStr = Lexer::getSourceText(
-        CharSourceRange::getTokenRange(Base->getSourceRange()),
-        Context.getSourceManager(), Context.getLangOpts());
-    Replacement = BaseStr.str() + ".last(" + ArgStr.str() + ")";
+        *Result.SourceManager, Result.Context->getLangOpts());
+
+    std::string Replacement =
+        SpanText.str() + ".first(" + CountText.str() + ")";
```

Comment:
Please use a twine instead of adding `std::string`s together: `(SpanTExt + ".first(" + CountText + ")").str()`